Automattic/pocket-casts-android
# Display playback progress during clip sharing

## Description

This PR adds playback progress to clip sharing. Playback progress is draggable. There are also some changes to the behavior.

* Play/pause button now pauses playback instead of stopping it.
* Changing start or end of a clip stops playback.

Note: Zooming is acting junky when timeline is not scrolled to the beginning. I plan to work on it in the future.

## Testing Instructions

1. Start sharing.
2. Drag progress to some position.
3. Play the clip.
4. Playback should start from the dragged position.
5. Let the clip playback to the end.
6. Play the clip again.
7. Playback should start.
8. Pause playback.
9. Playback should stop.
10. Resume playback.
11. Playback should start from the paused position.
12. While clip is playing drag end or start of the clip.
13. Playback should stop.
14. Resume playback.
15. Playback should start from the beginning.

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/30936061/20a73f2d-d12b-452e-aa12-798030bd9434

## Checklist
 
#### I have tested any UI changes...
